src/read/dwarf.rs

```diff
@@ -629,6 +633,11 @@ impl<R: Reader> Unit<R> {
             )?),
             None => None,
         };
+        if let Some(low_pc_attr) = low_pc_attr {
+            if let Some(addr) = dwarf.attr_address(&unit, low_pc_attr)? {
```

Comment:
Note: to be consistent, I followed example of `unit.{name,comp_dir,line_program}` above where invalid attribute form is silently ignored, but I'm not sure why it's done that way...

Wouldn't it be better to return explicit `UnsupportedAttributeForm` error in all those cases?

Reply:
Hmm, not sure either. I guess personally I would turn those fields into private OnceCells, and expose fallible getters as functions for them instead - that way, user could both handle an error, and not sacrifice any performance because variable would be cached once retrieved.

However, this would be a breaking API change, so probably not acceptable?

Reply:
Not sure I properly understand what you mean. Using OnceCell, retrieving the variable would need to iterate the attributes again in order to initialize the OnceCell?

Breaking changes are fine if we are sure it is the correct thing to do (but for this case I'm not sure). We still don't support all of DWARF 5, and that is likely to require more breaking changes.

Reply:
> retrieving the variable would need to iterate the attributes again in order to initialize the OnceCell

Yeah, API-wise it would be equivalent of `DebuggingInformationEntry::attr(name)` which already does that, but here it would do it with caching. As you said yourself, many consumers are already fine with ignoring some fields or errors in them.

I think most consumers don't actually need to access _all_ of the `Unit` properties, and emitters usually don't emit many properties on Unit node, so iteration won't have any noticeable performance impact.

Reply:
As another alternative, you could keep current iteration, but store `Result`s in private fields, and clone them in such getters. But I really think that making field accessors lazy makes more sense for common scenarios, as it will reduce costs for parsing things user doesn't care about.